src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
```

Comment:
You can simplify this method using ternary operator to improve readability:
```
    private Object[] grow() {
        int oldCapacity = array.length;
        double multiplicator = oldCapacity == size ? DEFAULT_MULTIPLICATOR : 1;
        if (oldCapacity > 0 || array != EMPTY_ARRAYDATA) {
            int newCapacity = (int) (oldCapacity * multiplicator + 1);
            return array = Arrays.copyOf(array, newCapacity);
        } else {
            return array = (T[]) new Object[DEFAULT_CAPACITY];
        }
    }
```

We can simplify it even more assuming that `oldCapacity > 0` is always `true`:
```
    private Object[] grow() {
        int oldCapacity = array.length;
        int newCapacity = oldCapacity == 0 ? DEFAULT_CAPACITY : (int) (oldCapacity * DEFAULT_MULTIPLICATOR + 1);
        return array = Arrays.copyOf(array, newCapacity);
    }
```